api/src/main/java/org/apache/gravitino/rel/indexes/Indexes.java

```diff
@@ -73,10 +89,81 @@ public static Index of(Index.IndexType indexType, String name, String[][] fieldN
         .build();
   }
 
+  /** Custom JSON serializer for Index objects. */
```

Comment:
It's not proper to add the JSON serde in the API module. Should if be in the class like `IndexDTO`, and then convert the `IndexDTO` to the IndexImpl here?